pkg/replicate/scheme.go

```diff
@@ -192,7 +198,8 @@ func (rs *replicationScheme) execute(ctx context.Context) error {
 	}
 
 	for id, meta := range metas {
-		if rs.blockFilter(meta) {
+		_, err := rs.fromBkt.ReaderWithExpectedErrs(rs.fromBkt.IsObjNotFoundErr).Get(ctx, path.Join(meta.ULID.String(), metadata.DeletionMarkFilename))
```

Comment:
Do we need to have `--ignore-deletion-marks-delay` instead? I wonder if it makes sense :thinking: Mhm, probably not, because we are only copying the blocks here, and they aren't being replaced with one another.

Also, do we need to log `err` if we fail (not due to 404) to check whether the deletion mark exists? Preferably with `debug` log level.

Reply:
Yep, I thought it's enough for replication just the fact that it's marked, didn't think of any use case for the delay here.

Debug log added here https://github.com/thanos-io/thanos/pull/5117/commits/69db471417ea9b6b9d6ef5baf8094da0f3d768cb